src/main/main.ts

```diff
@@ -359,6 +362,7 @@ function initializeIpc() {
 
   ipcMain.handle("execute launcher update", async (event) => {
     if (appUpdaterInstance === null) throw Error("appUpdaterInstance is null");
+    setV2Quitting(true);
```

Comment:
This is the part that keep quitAndInstall in appUpdaterInstance quit properly,
and I think this is the only entry to trigger quitAndInstall directly. 
but if I'm missing anything. please nudge me.